app/uk/gov/hmrc/economiccrimelevyaccount/models/integrationframework/FinancialDataResponse.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyaccount.models.integrationframework
+
+import play.api.http.Status._
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+case class FinancialDataResponse(totalisation: Option[Totalisation], documentDetails: Option[Seq[DocumentDetails]])
+
+object FinancialDataResponse {
+
+  implicit object FinancialDataResponseReads
+      extends HttpReads[Either[FinancialDataErrorResponse, FinancialDataResponse]] {
+    override def read(
+      method: String,
+      url: String,
+      response: HttpResponse
+    ): Either[FinancialDataErrorResponse, FinancialDataResponse] = {
+      val validationResult = response.json.validate[FinancialDataErrorResponse]
```

Comment:
How about combining the _error_ cases in **response.status** into one case, and then putting the response.json.validate[FinancialDataErrorResponse] in there, so that we don't have to validate the error on invocations when the status is OK.

Reply:
Combined all of the error codes into single check.